windows-gaming-driver/src/config.rs

```diff
@@ -82,19 +186,39 @@ impl From<DeviceId> for (u16, u16) {
 
 impl Config {
     pub fn save<P: AsRef<Path>>(&self, path: P) {
-        let contents = toml::to_string(self).unwrap();
+        let contents = serde_yaml::to_string(self).unwrap();
         let mut file = OpenOptions::new().create(true).write(true)
             .truncate(true).open(path).expect("Failed to open config file");
-        write!(file, "{}", contents).expect("Failed to write config file");
+        writeln!(file, "{}", contents).expect("Failed to write config file");
     }
 
-    pub fn load<P: AsRef<Path>>(path: P) -> Config {
+    pub fn load<P: AsRef<Path>>(path: P) -> Option<Config> {
```

Comment:
How about no? This sure seems like a smart idea ... until you realize that you basically have to fully duplicate the entire config code for every single version plus conversion code that doesn't even make sense in many corner cases (you had no PCI slots? well yeah good luck finding them).

Honest guess is something around 10x the code for config? And around 20x the work as we can no longer just (de)serialize and be done?

And what do we gain? A user base of virtually no one no longer has to deal with breaking changes to the config of a v0.1 tool.

The proper solution is to just not have any breaking changes. But it doesn't matter right now as we don't have a stable release anyways.

Sure I'm not opposed to doing it like this but I won't be the one developing **and maintaining** it.